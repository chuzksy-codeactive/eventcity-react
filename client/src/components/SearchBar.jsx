import React from 'react';

const SearchBar = () => {
  return (
    <section className="section">
      <div className="container-fluit searchbar">
        <div className="row justify-content-center">
          <div className="col-6">
            <form id="search" className="form-inline">
              <input
                className="form-control"
                id="search-input"
                type="text"
                name="search"
                placeholder="Search centers by name, facilities or location"
              />
              <button className="btn btn-primary" form="search" value="Submit">
                Search
              </button>
            </form>
          </div>
        </div>
      </div>
      <div>
        <h1 className="header-section row-width">Explore our centers and make a choice</h1>
        <p className="para-section row-width">
          We guide you from the stressful decision making process, ensuring you are comfortable,
          whether it is a wedding, corporate {`function`} or even a kiddies party, we create a buzz
          around you, taking you to the next level.
        </p>
      </div>
    </section>
  );
};
export default SearchBar;
