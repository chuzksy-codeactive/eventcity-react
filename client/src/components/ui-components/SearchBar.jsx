import React, { Component } from 'react';

/**
 * This component is used for render
 * search functionality in other component
 *
 * @function
 * @return {object} JSX DOM
 */

class SearchBar extends Component {
  render() {
    return (
      <section className="section">
        <div className="container-fluid searchbar">
          <div className="row justify-content-center">
            <div className="col-6">
              <input
                className="form-control inner-addon left-addon"
                id="search-input"
                type="text"
                name="search"
                placeholder="Search centers by name, facilities or location"
                onChange={this.props.onInputChange}
              />
            </div>
          </div>
        </div>
        <div>
          <h1 className="header-section row-width">Explore our centers and make a choice</h1>
          <p className="para-section row-width">
            We guide you from the stressful decision making process, ensuring you are comfortable, whether it is a wedding, corporate {'function'} or
            even a kiddies party, we create a buzz around you, taking you to the next level.
          </p>
        </div>
      </section>
    );
  }
}
export default SearchBar;
