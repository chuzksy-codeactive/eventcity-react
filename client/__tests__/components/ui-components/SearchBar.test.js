import React from 'react'
import SearchBar from '../../../src/components/ui-components/SearchBar'

describe('SearchBar Component test', () => {
  const props = {
    onInputChange: jest.fn()
  }
  it('It should render the component without error', () => {
    const wrapper = renderer.create(<SearchBar {...props}/>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
})