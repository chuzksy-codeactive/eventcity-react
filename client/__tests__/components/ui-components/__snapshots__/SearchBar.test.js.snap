// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component test It should render the component without error 1`] = `
<section
  className="section"
>
  <div
    className="container-fluid searchbar"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-6"
      >
        <input
          className="form-control inner-addon left-addon"
          id="search-input"
          name="search"
          onChange={[MockFunction]}
          placeholder="Search centers by name, facilities or location"
          type="text"
        />
      </div>
    </div>
  </div>
  <div>
    <h1
      className="header-section row-width"
    >
      Explore our centers and make a choice
    </h1>
    <p
      className="para-section row-width"
    >
      We guide you from the stressful decision making process, ensuring you are comfortable, whether it is a wedding, corporate 
      function
       or even a kiddies party, we create a buzz around you, taking you to the next level.
    </p>
  </div>
</section>
`;
